spec/13-functions.md

```diff
@@ -77,11 +77,11 @@ return-type:
   ':' 'void'
 
 type-declaration:
-  'array'
-  'callable'
-  'iterable'
-  scalar-type
-  qualified-name
+  '?'? 'array'
+  '?'? 'callable'
+  '?'? 'iterable'
+  '?'? scalar-type
+  '?'? qualified-name
```

Comment:
I think it would be better to split this in a way that does not require repeating the nullable prefix five times. For example

```
type-declaration:
  '?'? simple-type-declaration

simple-type-declaration:
  ...
```